src/infrastructure/channels/mod.rs

```diff
@@ -22,5 +22,5 @@ pub trait Channel {
     ) -> NetworkResult<PacketData>;
 
     /// Progress an packet on receive and receive the processed data.
```

Comment:
Since you changed this to option can you add a doc comment that explains when an option value will be `Some` or `None`.